---
bool success
string[] joint
float64[] lower_limit
float64[] upper_limit